apps/avatax/src/modules/avatax/avatax-sdk-client-factory.ts

```diff
@@ -0,0 +1,54 @@
+import Avatax from "avatax";
+import { LogOptions } from "avatax/lib/utils/logger";
+import packageJson from "../../../package.json";
+
+type AvataxSettings = {
+  appName: string;
+  appVersion: string;
+  environment: "sandbox" | "production";
+  machineName: string;
+  timeout: number;
+  logOptions?: LogOptions;
+};
+
+const defaultAvataxSettings: AvataxSettings = {
+  appName: packageJson.name,
+  appVersion: packageJson.version,
+  environment: "sandbox",
+  machineName: "tax-app",
+  timeout: parseInt(process.env.AVATAX_CLIENT_TIMEOUT ?? "15000", 10),
+};
+
+const createAvataxSettings = ({ isSandbox }: { isSandbox: boolean }): AvataxSettings => {
+  const settings: AvataxSettings = {
+    ...defaultAvataxSettings,
+    environment: isSandbox ? "sandbox" : "production",
+  };
+
+  return settings;
+};
+
+export interface IAvataxSdkClientFactory {
+  createClient(opts: {
+    isSandbox: boolean;
+    credentials: {
+      username: string;
+      password: string;
+    };
+  }): Avatax;
+}
+
+export class AvataxSdkClientFactory implements IAvataxSdkClientFactory {
+  createClient(config: {
```

Comment:
question: should this method be static?

Reply:
I'm not sure. Static methods work well as a factory method for the class. Factories can also be interchangeable. Eg. I can 
1: inject something to factory constructor, if my client creation requires dependency
2. Use interface and implement eg. MockFactory.

With static method you can easily mix static and non static with interface